src/threadpool-object.h

```diff
@@ -79,6 +79,31 @@ struct PTHREADPOOL_CACHELINE_ALIGNED thread_info {
 	 */
 	HANDLE thread_handle;
 #endif
+
```

Comment:
I don't see why all changes in this file are needed. Please revert.

Reply:
So this PR adds per thread wakeup logic so that only participating threads are woken up. That is why this change was needed. If you have better suggestions I am open to it.